packages/core/src/amazonqFeatureDev/util/files.ts

```diff
@@ -137,9 +153,9 @@ export async function prepareRepoData(
     span: Metric<AmazonqCreateUpload>
 ) {
     try {
+        const files = await collectFiles(repoRootPaths, workspaceFolders, true)
         const zip = new AdmZip()
```

Comment:
the current approach used to zip files does not "stream" the zip creation so will also run into memory constraints.

please look at https://github.com/aws/aws-toolkit-vscode/pull/4769 , can you rewrite this feature on top of that (after it lands)?

Reply:
What's the ETA of this? Presume we want both these in the same wave of the toolkit?

Reply:
Yeah I'm happy to change our implementation to use the stream after it's merged.